store/store.go

```diff
@@ -421,29 +422,56 @@ func (bs *BlockStore) SaveSeenCommit(height int64, seenCommit *types.Commit) err
 	if err != nil {
 		return fmt.Errorf("unable to marshal commit: %w", err)
 	}
-	return bs.db.Set(calcSeenCommitKey(height), seenCommitBytes)
+	return bs.db.Set(seenCommitKey(height), seenCommitBytes)
 }
 
-//-----------------------------------------------------------------------------
+//---------------------------------- KEY ENCODING -----------------------------------------
+
+// key prefixes
+const (
+	// prefixes are unique across all tm db's
+	prefixBlockMeta   = byte(0x00)
+	prefixBlockPart   = byte(0x01)
+	prefixBlockCommit = byte(0x02)
+	prefixSeenCommit  = byte(0x03)
+	prefixBlockHash   = byte(0x04)
+)
 
-func calcBlockMetaKey(height int64) []byte {
-	return []byte(fmt.Sprintf("H:%v", height))
+func blockMetaKey(height int64) []byte {
+	key, err := orderedcode.Append([]byte{prefixBlockMeta}, height)
+	if err != nil {
+		panic(err)
+	}
+	return key
 }
 
-func calcBlockPartKey(height int64, partIndex int) []byte {
-	return []byte(fmt.Sprintf("P:%v:%v", height, partIndex))
+func blockPartKey(height int64, partIndex int) []byte {
+	key, err := orderedcode.Append([]byte{prefixBlockPart}, height, uint64(partIndex))
+	if err != nil {
+		panic(err)
+	}
+	return key
 }
 
-func calcBlockCommitKey(height int64) []byte {
-	return []byte(fmt.Sprintf("C:%v", height))
+func blockCommitKey(height int64) []byte {
+	key, err := orderedcode.Append([]byte{prefixBlockCommit}, height)
```

Comment:
Does `int64` and `uint64` encode differently? If so, we may want to pre-emptively use `uint64` for heights, since I think we'll want to migrate to that in the future.

Reply:
Yeah they are encoded differently. Strangely int64 is more efficient than uint64 in terms of size

Reply:
```height: 1000000 int64: [11101111 1000010 1000000] (len: 3), uint64: [11 1111 1000010 1000000] (len: 4)```

Reply:
Probably because `int` is used more often than `uint`, so they're optimizing for the typical case. Also, `1000000` is a fairly large value, I'm guessing there's no overhead for smaller values.

Reply:
uint always has an extra byte for the length so there is always overhead. signed int has a different strategy where the length is built in to the first byte (n amount of 1's followed by a 0 indicating the length of bytes to come) making it generally more efficient. I don't know why uint couldn't have followed the same strategy

Reply:
Seems so, that's unfortunate. 😞